components/cdc/src/delegate.rs

```diff
@@ -346,10 +375,23 @@ impl Delegate {
         info!("region met error";
             "region_id" => self.region_id, "error" => ?err);
         let change_data_err = self.error_event(err);
-        for d in &self.downstreams {
-            d.state.store(DownstreamState::Stopped);
+
+        for d in self.downstreams() {
+            {
+                let mut incremental_state = d.incremental_scan_state.lock().unwrap();
+                match *incremental_state {
+                    IncrementalScanState::Ongoing => {
+                        info!("cdc incremental scan not done, holding off error"; "region_id" => self.region_id);
+                        *incremental_state =
+                            IncrementalScanState::ErrorPending(change_data_err.clone())
+                    }
+                    _ => {
+                        d.state.store(DownstreamState::Stopped);
+                        d.sink_error(change_data_err.clone());
+                    }
+                }
+            }
```

Comment:
Unnecessary indentation?

```suggestion
            let mut incremental_state = d.incremental_scan_state.lock().unwrap();
            match *incremental_state {
                IncrementalScanState::Ongoing => {
                    info!("cdc incremental scan not done, holding off error"; "region_id" => self.region_id);
                    *incremental_state =
                        IncrementalScanState::ErrorPending(change_data_err.clone())
                }
                _ => {
                    d.state.store(DownstreamState::Stopped);
                    d.sink_error(change_data_err.clone());
                }
            }
```